File: Bookcase core/src/main/java/cz/vhromada/book/entity/Category.java
package cz.vhromada.book.entity;

import java.util.Objects;

import cz.vhromada.common.Movable;

/**
 * A class represents category.
 *
 * @author Vladimir Hromada
 */
public class Category implements Movable {

    /**
     * SerialVersionUID
     */
    private static final long serialVersionUID = 1L;

    /**
     * ID
     */
    private Integer id;

    /**
     * Name
     */
    private String name;

    /**
     * Position
     */
    private Integer position;

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(final Integer id) {
        this.id = id;
    }

    /**
     * Returns name.
     *
     * @return name
     */
    public String getName() {
        return name;
    }

    /**
     * Sets a new value to name.
     *
     * @param name new value
     */
    public void setName(final String name) {
        this.name = name;
    }

    @Override
    public Integer getPosition() {
        return position;
    }

    @Override
    public void setPosition(final Integer position) {
        this.position = position;
    }

    @Override
    public boolean equals(final Object obj) {
        if (this == obj) {
            return true;
        }

        if (!(obj instanceof Category) || id == null) {
            return false;
        }

        return id.equals(((Category) obj).id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

    @Override
    public String toString() {
        return String.format("Category [id=%d, name=%s, position=%d]", id, name, position);
    }

}

File: bookcase-core/src/main/kotlin/cz/vhromada/bookcase/entity/Category.kt
package cz.vhromada.bookcase.entity

import cz.vhromada.common.entity.Movable
import java.util.Objects

/**
 * A class represents category.
 *
 * @author Vladimir Hromada
 */
data class Category(
        /**
         * ID
         */
        override var id: Int?,

        /**
         * Name
         */
        val name: String?,

        /**
         * Position
         */
        override var position: Int?) : Movable {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        return if (other !is Category || id == null) {
            false
        } else id == other.id
    }

    override fun hashCode(): Int {
        return Objects.hashCode(id)
    }

    override fun toString(): String {
        return String.format("Category [id=%d, name=%s, position=%d]", id, name, position)
    }

}
